src/elements/AddArtifactTagPopUp/AddArtifactTagPopUp.js

```diff
@@ -52,6 +55,15 @@ const AddArtifactTagPopUp = ({
     })
   )
   const location = useLocation()
+  const { handleCloseModal, resolveModal } = useModalBlockHistory(onResolve, formRef.current)
+
+  useEffect(() => {
+    onInit &&
+      onInit().then(results => {
+        const tags = results.filter(result => result.tag).map(result => result.tag)
+        setExistingTags(tags)
+      })
```

Comment:
`onInit` isn't an appropriate name for this prop. Because we can pass any function. Only the function that gets an Artifact